src/http.rs

```diff
@@ -95,29 +130,52 @@ impl Client {
 }
 
 pub struct Download {
-    resp: reqwest::Response,
+    url: String,
+    client: Client,
     timeout: Option<Duration>,
     pub progress: u64,
     pub total: u64,
 }
 
 impl Download {
     pub async fn chunk(&mut self) -> Result<Option<Bytes>> {
-        let future = self.resp.chunk();
-        let bytes = if let Some(timeout) = self.timeout {
-            if let Ok(bytes) = time::timeout(timeout, future).await {
-                bytes?
+        let begin = self.progress;
+        let end = (begin + (self.total / 10)).min(self.total);
+
+        if self.total <= self.progress {
+            debug!("finished!");
+            return Ok(None);
+        }
+        info!("fetching range {begin} => {end}");
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            RANGE,
+            HeaderValue::from_str(&format!("bytes={begin}-{end}"))?,
+        );
+        let future = self.client.client.get(&self.url).headers(headers).send();
```

Comment:
The `chunk` method should probably not restart the download, instead in case of an error the download should be restarted in `download_pkg`.